tooling/noir_js/test/backend/barretenberg.ts

```diff
@@ -0,0 +1,122 @@
+// @ts-ignore
+import { Barretenberg, Crs, RawBuffer } from '@aztec/bb.js';
+// TODO: This should be re-exported from @aztec/bb-js
+import { Ptr } from '@aztec/bb.js/dest/browser/types';
+import { acirToUint8Array } from '../../src/index.js';
+
+export class Backend {
+    // These type assertions are used so that we don't
+  // have to initialize `api` and `acirComposer` in the constructor.
+  // These are initialized asynchronously in the `init` function,
+  // constructors cannot be asynchronous which is why we do this.
+  api = {} as Barretenberg;
+  acirComposer = {} as Ptr;
+  acirUncompressedBytecode: Uint8Array;
+
+  constructor(acirBytecodeBase64: string) {
+    this.acirUncompressedBytecode = acirToUint8Array(acirBytecodeBase64);
+  }
+  async init() {
```

Comment:
This file will likely move into bb.js or another package, so that non-advanced users don't need to know about slab allocators etc. We can make numThreads a property of this class and pass it in via the constructor with a default